ddtrace/bootstrap/sitecustomize.py

```diff
@@ -203,15 +203,12 @@ def _(threading):
         env_tags = os.getenv("DD_TRACE_GLOBAL_TAGS")
         tracer.set_tags(parse_tags_str(env_tags))
 
-    if sys.version_info >= (3, 7) and asbool(os.getenv("DD_TRACE_OTEL_ENABLED", False)):
+    if sys.version_info >= (3, 7) and config._otel_enabled:
         from opentelemetry.trace import set_tracer_provider
 
         from ddtrace.opentelemetry import TracerProvider
 
         set_tracer_provider(TracerProvider())
-        # Replaces the default otel api runtime context with DDRuntimeContext
-        # https://github.com/open-telemetry/opentelemetry-python/blob/v1.16.0/opentelemetry-api/src/opentelemetry/context/__init__.py#L53
-        os.environ["OTEL_PYTHON_CONTEXT"] = "ddcontextvars_context"
```

Comment:
does the order of this matter?

Reply:
we would be delaying to later, no?

Reply:
Order doesn't really matter as long as this environment variable is set before the first otel span is generated.
This will actually occur earlier since the config object is one of the first things imported by ddtrace.

Reply:
what happens if we change this env var after the first span is created?

when does OTel look at this env variable?

Reply:
Nope. In that scenario a noop tracer provider is set and spans are not sent. 

A warning log is generated in this scenario. It's easy to debug.